app/v1/validation/fullReturn/FullReturnValidator.scala

```diff
@@ -127,17 +127,29 @@ trait FullReturnValidator extends BaseValidation {
     }
   }
 
-  private def validateNoTotalIncomeWhenRestriction: ValidationResult[BigDecimal] = {
+  private def validateAggInterestAndReallocationOrRestrictionStatus: ValidationResult[BigDecimal] = {
     val aggregateNetTaxInterest: BigDecimal = fullReturnModel.aggregateNetTaxInterest
     val subjectRestrictions: Boolean = fullReturnModel.groupSubjectToInterestRestrictions
+    val subjectReactivation: Boolean = fullReturnModel.groupSubjectToInterestReactivation
 
-    if (aggregateNetTaxInterest > 0 && subjectRestrictions) {
-      NoTotalIncomeWhenSubjectToRestriction(aggregateNetTaxInterest, subjectRestrictions).invalidNec
-    } else {
-      aggregateNetTaxInterest.validNec
+    (aggregateNetTaxInterest, subjectRestrictions, subjectReactivation) match {
+      case (x, y, z) if x < 0 && !y && z => AggInterestNegativeAndReactivation(aggregateNetTaxInterest, subjectReactivation).invalidNec
+      case (x, y, z) if x > 0 && y && !z => AggInterestPositiveAndRestriction(aggregateNetTaxInterest, subjectRestrictions).invalidNec
+      case _ => aggregateNetTaxInterest.validNec
+    }
+  }
+
+  private def validateReactivationCapOnlyIfSubjectReactivation: ValidationResult[Boolean] = {
+    val subjectReactivation: Boolean = fullReturnModel.groupSubjectToInterestReactivation
+    val reactivationCap:Boolean = fullReturnModel.groupLevelAmount.interestReactivationCap.isDefined
+
+    (subjectReactivation, reactivationCap) match {
+      case (true, true) => reactivationCap.validNec
+      case _ => ReactivationCapAndNotSubjectToReactivation().invalidNec
```

Comment:
Wouldn't `case (false, false)` also be valid? At the moment that would be flagged as invalid.